### PR TITLE
[WIP] Fix for timeout in graph_model

### PR DESCRIPTION
**Pull request by rpgoldman**
See #3458

### CODE REVIEW

**Comment by rpgoldman** [comment]
This PR is *almost* right, I think, but somehow by changing the ancestor computation, I have changed the computation of the dimensions in graph rendering.

File: pymc3/model_graph.py [hunk 1]
```diff
@@ -40,24 +47,38 @@ def _ancestors(self, var, func, blockers=None):
...
-        deterministic as a blocker, it will skip those nodes.
+        """Get all ancestors of a function, doing some accounting for deterministics.
         """
-        deterministics = self.get_deterministics(var)
+
+        # this contains all of the variables in the model EXCEPT var...
+        vars: List[var] = set(self.var_list)
+        vars.remove(var)
+        
         upstream = self._ancestors(var, func)
```
Comment (lucianopaz) [inline pymc3/model_graph.py]: I would completely remove `self._ancestors` and replace it with the `stack_search` that you did

**Review by lucianopaz — CHANGES_REQUESTED** [review]
Given that you observed a change in the computed dimensions, could you add a new test to the test suite that checks if the computed dimensions match with what should be returned?

**Comment by rpgoldman** [comment]
@lucianopaz actually the dimension change was not an issue in the breadth first search -- it revealed an error in the code I used to debug the search! That's a relief. I will see if I can propose some tests, but I don't understand the test suite  very well.
I will try to make the change you suggest today.

**Comment by rpgoldman** [comment]
@lucianopaz I have just looked over the existing tests for the model grapher, and they check the structure, including the identification of plates.  So I think they are sufficient in their present form.  There is no obvious test to add based on the fix, because the old version didn't compute the wrong thing; it simply didn't complete. But I am open to suggestions.

Note: if this PR looks good, I should squash it before it's merged.

File: pymc3/model_graph.py [hunk 2]
```diff
@@ -40,24 +47,32 @@ def _ancestors(self, var, func, blockers=None):
         return set([j for j in ancestors([func], blockers=blockers) if j in self.var_list and j != var])
```
Comment (lucianopaz) [inline pymc3/model_graph.py]: `_ancestors` now seems to be unused, right? If so, it should be removed.

**Review by lucianopaz — CHANGES_REQUESTED** [review]
Everything looks fine to me. No other tests should be needed. Once `_ancestors` is removed, just add a line to the release notes under maintenance and the PR should be ready to merge

**Comment by rpgoldman** [comment]
@lucianopaz OK, I think this is done now.

**Comment by lucianopaz** [comment]
Great! Thanks a lot @rpgoldman! Once the tests pass, I'll merge

**Review by lucianopaz — APPROVED** [review]

**Review by lucianopaz — CHANGES_REQUESTED** [review]
Oh, I had forgotten that you still need to add a line to RELEASE-NOTES.md under the maintenance section referencing the fixed issue

**Comment by rpgoldman** [comment]
@lucianopaz Pushed the documentation and removed an unused import causing the check failure.

**Comment by lucianopaz** [comment]
Thanks again @rpgoldman!

**Comment by rpgoldman** [comment]
@lucianopaz No problem! I'm glad to be able to give even a little something back in exchange for all I've gotten from the community!

**Comment by twiecki** [comment]
Thanks @rpgoldman!

**Pull request merged**